etcd_helper/etcdfuncs.go

```diff
@@ -0,0 +1,113 @@
+package etcd_helper
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+
+	"github.com/coreos/etcd/client"
+)
+
+type Etcdreader struct {
+	etcdServiceURL string
+}
+type Etcdwriter struct {
+	etcdServiceURL string
+}
+
+func (o Etcdreader) get(resourceKey string) (string, error) {
+	cfg := client.Config{
+		Endpoints: []string{o.etcdServiceURL},
+		Transport: client.DefaultTransport,
+	}
+	c, err := client.New(cfg)
+	if err != nil {
+		log.Fatal(err)
+	}
+	kapi := client.NewKeysAPI(c)
+
+	resp, err1 := kapi.Get(context.Background(), resourceKey, nil)
+	if err1 != nil {
+		return "", fmt.Errorf("Error: %v", err1)
+	} else {
+		//log.Printf("%q key has %q value\n", resp.Node.Key, resp.Node.Value)
+	}
+	return resp.Node.Value, nil
+}
+
+func (o Etcdreader) getList(resourceKey string) ([]string, error) {
+	cfg := client.Config{
+		Endpoints: []string{o.etcdServiceURL},
+		Transport: client.DefaultTransport,
+	}
+	c, err := client.New(cfg)
+	if err != nil {
+		return nil, fmt.Errorf("Error: %v", err)
```

Comment:
We should have consistency for return values in get() and getList().

In get() there are places where we are returning empty string ('') on error. Above we are returning nil. I think we should return nil in get() also.

Reply:
In get() there is still "" as return when error occurs. We can change that to nil